src/ark/utils/masking_utils.py

```diff
@@ -0,0 +1,123 @@
+import os
+import numpy as np
+
+from alpineer import io_utils, misc_utils, load_utils
+from ark.utils import data_utils
+from ark.segmentation.ez_seg.composites import composite_builder
+from ark.segmentation.ez_seg.ez_object_segmentation import _create_object_mask
+
+
+def generate_signal_masks(img_dir, mask_dir, channels, mask_name, intensity_thresh_perc="auto",
+                          sigma=2, min_mask_size=5000, max_hole_size=1000):
+    """Creates a single signal mask for each FOV when given the channels to aggregate.
+
+    Args:
+        img_dir (str): path to the image tiff directory
+        mask_dir (str): path where the masks will be saved
+        channels (list): list of channels to combine to create a single mask for
+        mask_name (str): name for the new mask file created
+        intensity_thresh_perc (int): percentile to threshold intensity values in the image,
+            defaults to "auto" which will calculate an appropriate percentile for the user
+        sigma (float): sigma for gaussian blur
+        min_mask_size (int): minimum size of masked objects to include
+        max_hole_size (int): maximum size of holes to leave in masked objects
+    """
+    # check correct image directory path
+    io_utils.validate_paths([img_dir])
+    fovs = io_utils.list_folders(img_dir)
+
+    # check valid channel name
+    channel_list = io_utils.remove_file_extensions(
+        io_utils.list_files(os.path.join(img_dir, fovs[0])))
+    misc_utils.verify_in_list(input_channels=channels, all_channels=channel_list)
+
+    # create composite image (or read in single image)
+    composite_imgs = composite_builder(
+        img_dir, img_sub_folder='', fov_list=fovs, images_to_add=channels, images_to_subtract=[],
+        image_type='total', composite_method='total')
+
+    for fov in fovs:
+        # create mask
+        img = composite_imgs[fov]
+        img_size = img.shape[0] * img.shape[1]
+        mask = _create_object_mask(img, 'blob', sigma, intensity_thresh_perc, max_hole_size,
+                                   fov_dim=400, min_object_area=min_mask_size,
+                                   max_object_area=img_size)
+
+        # save mask
+        save_dir = os.path.join(mask_dir, fov)
+        if not os.path.exists(save_dir):
+            os.makedirs(save_dir)
```

Comment:
Really minor but I've started to see this enough that I think the minor simplification of `os.makedirs(save_dir, exist_ok=True)` is better.